doc/Type/independent-routines.pod6

```diff
@@ -634,6 +634,34 @@ token. Directives guide the use (if any) of the arguments. When a directive
 other than C<%> is used, it indicates how the next argument passed is to be
 formatted into the string to be created.
 
+=begin head3 Warning
+
+The C<$format> may be defined enclosed in single or double
+quotes. The double-quoted C<$format> string is interpolated before
+being scanned and any embedded string whose interpolated value
+contains a '%' character will cause an exception. For example:
+
+=begin code
+my $prod = "Ab-%x-42";
+my $cost = "30";
+sprintf("Product $prod; cost: \$%d", $cost).put;
+Your printf-style directives specify 2 arguments, but 1 argument was supplied
+  in block <unit> at <unknown file> line 1
+=end code
+
+When handling unknown input you should avoid using such syntax by
+putting all variables in the C<*@args> array and have one '%' for each
+in C<$format>. If you need to include a '$' symbol in the format
+string either escape it or use the single-quoted form.  For example,
+either form works without error:
+
+=begin code
+sprintf("2 x \$20 = \$%d", 2*20).put; # OUTPUT: «2 x $20 = $40␤»
+sprintf('2 x $20 = $%d', 2*20").put;  # OUTPUT: «2 x $20 = $40␤»
+=end code
+
+=end head3
+
```

Comment:
Also, this is going to probably mess with indexing. If what you want to achieve is somehow to highlight it, It's probably better to do it some other way...

Reply:
Yeah, I knew better. Suggestions welcome!

Reply:
Big bold italics sign at the top saying "Dangerous waters ahead"?